src/test/java/io/confluent/examples/streams/CustomStreamTableJoinIntegrationTest.java

```diff
@@ -72,6 +72,10 @@
  *
  * Note: This example works with Java 8+ only.
  *
+ * TODO (and note for maintainers): This test can be migrated to use the TopologyTestDriver of Kafka Streams only once
+ * the driver is handling timestamps exactly like a "real" setup does. Until then, this test must use an embedded
```

Comment:
What are the differences in how TTD handles timestamps?

Reply:
IIRC the difference is that TTD processes each event synchronously in exactly the order you pass them, whereas KStreams buffers both inputs (of a join) and selects records based on the lowest timestamp. @vvcephei knows more.

Reply:
This is correct. The comment implies this is a flaw and maybe that there are plans to change it, but AFAIK, there are no plans to change it, and I'd tentatively say it's a feature, not a bug.

If TTD behaved _exactly_ like a real setup in this regard, it would be just as difficult to write reliable tests as it is against a real setup.

Specifically in regard to whether this test can be migrated or not, would it not work to just pipe the inputs in the same order as their timestamps?